src/inference/src/cpp/infer_request.cpp

```diff
@@ -108,6 +109,24 @@ void InferRequest::set_input_tensor(size_t idx, const Tensor& tensor) {
                         " was not found! The model has only ",
                         inputs.size(),
                         " inputs.");
+        auto device_name = _impl->get_compiled_model()->get_context()->get_device_name();
```

Comment:
As it was suggested in [this comment](https://github.com/openvinotoolkit/openvino/pull/26996#pullrequestreview-2365801166) the changes should be plugin specific as there is no issue with other plugins. 
There should be no changes in common sources (`Core`) part.